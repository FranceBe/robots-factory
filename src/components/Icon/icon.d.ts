import { FontAwesomeIconProps } from '@fortawesome/react-fontawesome'
import { IconEnum } from 'utils/common.enum'

export type IconProps = Partial<FontAwesomeIconProps> & {
  type?: IconEnum
}
