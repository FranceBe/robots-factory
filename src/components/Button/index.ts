import ButtonDefault, { Button } from 'components/Button/Button.component'

export { Button }

export default ButtonDefault
