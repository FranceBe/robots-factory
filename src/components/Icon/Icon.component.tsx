// We are using Font Awesome to display all the icons we need.
// By using the prop "type" we can chose which font awesome icon to display
import { IconProp } from '@fortawesome/fontawesome-svg-core'
import { faCheck, faRobot, faSpinner, faTimes } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { IconProps, IconType } from 'components/Icon/types/icon'
import React from 'react'

export const Icon: React.FC<IconProps> = ({ type = 'robot', ...rest }) => {
  const FAIconByIconProps: Record<IconType, IconProp> = {
    failure: faTimes,
    robot: faRobot,
    spinner: faSpinner,
    success: faCheck,
  }

  return <FontAwesomeIcon {...rest} icon={FAIconByIconProps[type]} />
}

export default Icon
