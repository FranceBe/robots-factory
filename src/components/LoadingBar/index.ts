import { LoadingBar } from 'components/LoadingBar/LoadingBar.component'

export { LoadingBar }
