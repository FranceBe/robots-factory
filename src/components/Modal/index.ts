import { Modal } from 'components/Modal/Modal.component'

export { Modal }
