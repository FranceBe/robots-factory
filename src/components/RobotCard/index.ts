import { RobotCard } from 'components/RobotCard/RobotCard.component'

export { RobotCard }
