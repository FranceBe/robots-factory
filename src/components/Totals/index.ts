import { Totals } from 'components/Totals/Totals.component'

export { Totals }
