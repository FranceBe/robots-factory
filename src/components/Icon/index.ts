import { FAIconByIconProps, Icon } from 'components/Icon/Icon.component'

export { FAIconByIconProps, Icon }
