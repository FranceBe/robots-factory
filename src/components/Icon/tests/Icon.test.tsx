// Since we use FontAwesome icon, we assume the FontAwesome module
// is tested by itself. To test our Icon component, we will only
// test that it renders correctly with the corresponding FontAwesome icon
import '@testing-library/jest-dom/extend-expect'

import { render, screen } from '@testing-library/react'
import { FAIconByIconProps, Icon } from 'components/Icon'
import { IconProps } from 'components/Icon/icon'
import React from 'react'

describe('Icon component', () => {
  const initTest = (props: IconProps) => render(<Icon {...props} />)
  describe('FAIconByIconProps', () => {
    it('should match snapshot', () => {
      expect(FAIconByIconProps).toMatchSnapshot()
    })
  })
  describe('Given type is "robot"', () => {
    it('should render the FontAwesome robot icon', () => {
      initTest({ type: 'robot' })
      const robotIcon = screen.getByRole('img', { hidden: true })
      expect(robotIcon).toBeInTheDocument()
      expect(robotIcon).toHaveClass('fa-robot')
    })
  })
  describe('Given type is "spinner"', () => {
    it('should render the FontAwesome spinner icon', () => {
      initTest({ type: 'spinner' })
      const spinnerIcon = screen.getByRole('img', { hidden: true })
      expect(spinnerIcon).toBeInTheDocument()
      expect(spinnerIcon).toHaveClass('fa-spinner')
    })
  })
  describe('Given type is "failure"', () => {
    it('should render the FontAwesome times icon', () => {
      initTest({ type: 'failure' })
      const failureIcon = screen.getByRole('img', { hidden: true })
      expect(failureIcon).toBeInTheDocument()
      expect(failureIcon).toHaveClass('fa-times')
    })
  })
  describe('Given type is "success"', () => {
    it('should render the FontAwesome check icon', () => {
      initTest({ type: 'success' })
      const successIcon = screen.getByRole('img', { hidden: true })
      expect(successIcon).toBeInTheDocument()
      expect(successIcon).toHaveClass('fa-check')
    })
  })
  describe('Given type is "moving"', () => {
    it('should render the FontAwesome check icon', () => {
      initTest({ type: 'moving' })
      const successIcon = screen.getByRole('img', { hidden: true })
      expect(successIcon).toBeInTheDocument()
      expect(successIcon).toHaveClass('fa-route')
    })
  })
})
