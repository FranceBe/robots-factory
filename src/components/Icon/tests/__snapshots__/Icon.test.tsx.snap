// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component FAIconByIconProps should match snapshot 1`] = `
Object {
  "failure": Object {
    "icon": Array [
      352,
      512,
      Array [],
      "f00d",
      "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
    ],
    "iconName": "times",
    "prefix": "fas",
  },
  "moving": Object {
    "icon": Array [
      512,
      512,
      Array [],
      "f4d7",
      "M416 320h-96c-17.6 0-32-14.4-32-32s14.4-32 32-32h96s96-107 96-160-43-96-96-96-96 43-96 96c0 25.5 22.2 63.4 45.3 96H320c-52.9 0-96 43.1-96 96s43.1 96 96 96h96c17.6 0 32 14.4 32 32s-14.4 32-32 32H185.5c-16 24.8-33.8 47.7-47.3 64H416c52.9 0 96-43.1 96-96s-43.1-96-96-96zm0-256c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zM96 256c-53 0-96 43-96 96s96 160 96 160 96-107 96-160-43-96-96-96zm0 128c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z",
    ],
    "iconName": "route",
    "prefix": "fas",
  },
  "robot": Object {
    "icon": Array [
      640,
      512,
      Array [],
      "f544",
      "M32,224H64V416H32A31.96166,31.96166,0,0,1,0,384V256A31.96166,31.96166,0,0,1,32,224Zm512-48V448a64.06328,64.06328,0,0,1-64,64H160a64.06328,64.06328,0,0,1-64-64V176a79.974,79.974,0,0,1,80-80H288V32a32,32,0,0,1,64,0V96H464A79.974,79.974,0,0,1,544,176ZM264,256a40,40,0,1,0-40,40A39.997,39.997,0,0,0,264,256Zm-8,128H192v32h64Zm96,0H288v32h64ZM456,256a40,40,0,1,0-40,40A39.997,39.997,0,0,0,456,256Zm-8,128H384v32h64ZM640,256V384a31.96166,31.96166,0,0,1-32,32H576V224h32A31.96166,31.96166,0,0,1,640,256Z",
    ],
    "iconName": "robot",
    "prefix": "fas",
  },
  "spinner": Object {
    "icon": Array [
      512,
      512,
      Array [],
      "f110",
      "M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z",
    ],
    "iconName": "spinner",
    "prefix": "fas",
  },
  "success": Object {
    "icon": Array [
      512,
      512,
      Array [],
      "f00c",
      "M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z",
    ],
    "iconName": "check",
    "prefix": "fas",
  },
}
`;
